### ISSUE TITLE
Getting "Loss not finite :(". 

### ISSUE THREAD

**Issue by cduguet** [darglein/ADOP#56]
Hello, I'm preparing a dataset of the interior of an appartment. I checked and COLMAP seems to work well, despite that there is a fair amount of white walls.  After a random number of epochs, I'm getting the following (pasting only the tail of the output):  

```
=== Epoch 81 ===

Train 81 |   0% |                              |    0/1016 [00:00:0000] [0.00 e/s] 
Train 81 |   0% |                              |    0/1016 [00:00:0000] [0.00 e/s] 
Train 81 |   4% |#                             |   40/1016 [00:05:0000] [8.00 e/s]  Cur=212.233490 Avg=302.391327
Train 81 |   9% |##                            |   88/1016 [00:10:0000] [8.80 e/s]  Cur=333.448730 Avg=287.600220
Train 81 |  13% |####                          |  136/1016 [00:15:0000] [9.07 e/s]  Cur=441.840668 Avg=295.512512
Train 81 |  18% |#####                         |  184/1016 [00:20:0000] [9.20 e/s]  Cur=379.657135 Avg=317.960571
Train 81 |  22% |######                        |  224/1016 [00:25:0000] [8.96 e/s]  Cur=277.145172 Avg=296.512878
Train 81 |  27% |########                      |  272/1016 [00:30:0000] [9.07 e/s]  Cur=310.707367 Avg=302.186951
Train 81 |  31% |#########                     |  320/1016 [00:35:0000] [9.14 e/s]  Cur=34.233173 Avg=287.412201
Train 81 |  36% |##########                    |  368/1016 [00:40:0001] [9.20 e/s]  Cur=33.885960 Avg=282.051544
Train 81 |  41% |############                  |  416/1016 [00:45:0001] [9.24 e/s]  Cur=267.568176 Avg=272.566742
Train 81 |  46% |#############                 |  464/1016 [00:50:0001] [9.28 e/s]  Cur=376.509857 Avg=253.497742
Train 81 |  50% |###############               |  512/1016 [00:55:0001] [9.31 e/s]  Cur=347.692017 Avg=255.870377
Train 81 |  55% |################              |  560/1016 [01:00:0001] [9.33 e/s]  Cur=222.076523 Avg=251.570724
Train 81 |  60% |#################             |  608/1016 [01:05:0001] [9.35 e/s]  Cur=184.181763 Avg=259.685944
Train 81 |  65% |###################           |  656/1016 [01:10:0002] [9.37 e/s]  Cur=30.161480 Avg=255.755127
Train 81 |  69% |####################          |  704/1016 [01:15:0002] [9.39 e/s]  Cur=34.554646 Avg=257.362579
Train 81 |  74% |######################        |  752/1016 [01:20:0002] [9.40 e/s]  Cur=239.453049 Avg=248.403091
Train 81 |  78% |#######################       |  792/1016 [01:25:0002] [9.32 e/s]  Cur=33.262161 Avg=248.473404
Train 81 |  83% |########################      |  840/1016 [01:30:0002] [9.33 e/s]  Cur=348.131836 Avg=255.551346
Train 81 |  87% |##########################    |  888/1016 [01:35:0003] [9.35 e/s]  Cur=547.986633 Avg=261.241058
Train 81 |  92% |###########################   |  936/1016 [01:40:0003] [9.36 e/s]  Cur=268.251221 Avg=261.505127
Tensor [8, 4, 512, 512] float cuda:0 Min/Max 1.3125e-06 20599.7 Mean 0.520595 Sum 4.36707e+06 sdev 7.16297 req-grad 1
Tensor [8, 4, 256, 256] float cuda:0 Min/Max 3.05836e-05 34.5411 Mean 0.535766 Sum 1.12358e+06 sdev 0.771836 req-grad 1
Tensor [8, 4, 128, 128] float cuda:0 Min/Max 1.22896e-05 33.5968 Mean 0.552693 Sum 289770 sdev 0.658228 req-grad 1
Tensor [8, 4, 64, 64] float cuda:0 Min/Max 0.000194362 36.972 Mean 0.557636 Sum 73090.4 sdev 0.61885 req-grad 1

Tensor [8, 2, 512, 512] float cuda:0 Min/Max -1 0.999414 Mean 0.083806 Sum 351508 sdev 0.473772 req-grad 0
Tensor [1] float cuda:0 Min/Max nan nan Mean nan Sum nan sdev nan req-grad 1
Tensor [8, 3, 512, 512] float cuda:0 Min/Max nan nan Mean nan Sum nan sdev nan req-grad 1
Tensor [1, 1, 512, 512] float cuda:0 Min/Max 0 1 Mean 0.878906 Sum 230400 sdev 0.326237 req-grad 0
Tensor [8, 3, 512, 512] float cuda:0 Min/Max 0 0.956863 Mean 0.653581 Sum 4.11197e+06 sdev 0.274293 req-grad 0

Scene:
Scene Log - Texture: Tensor [4, 11655018] float cuda:0 Min/Max -22.6205 15.4725 Mean -0.8064 Sum -3.75944e+07 sdev 1.10892 req-grad 1
Background Desc:  -0.970885 -0.863956 -0.900406 -0.882389 
Environment map: Tensor [1, 4, 1024, 512] float cuda:0 Min/Max -6.2707 3.55126 Mean -0.982767 Sum -2.06101e+06 sdev 0.274843 req-grad 1
Poses: Tensor [254, 8] double cuda:0 Min/Max -6.83462 8.02766 Mean 0.196939 Sum 400.181 sdev 1.62054 req-grad 0
Vignette params:  -0.0149647 -0.00850714 -0.00492687 |   0.00124693 -0.000540384
terminate called after throwing an instance of 'std::runtime_error'
  what():  Loss not finite :(
```

Could it be due to the lack of enough points on the white walls? If so, how would you go about it? I'm willing to provide any data if required to better assess this.

**Comment by darglein** [comment]
Hi, please post your training config file.

**Comment by cduguet** [comment]
Thank you! Here it is:

```
[TrainParams]
random_seed = 3746934646
do_train = true
do_eval = true
batch_size = 2
inner_batch_size = 4
inner_sample_size = 3
num_epochs = 600
save_checkpoints_its = 10
eval_only_on_checkpoint = true
name = josefstaedterstrasse64
debug = false
output_file_type = .jpg
split_method = 
max_images = -1
duplicate_train_factor = 1
shuffle_initial_indices = false
shuffle_train_indices = true
split_remove_neighbors = 0
split_index_file_train = 
split_index_file_test = 
train_on_eval = true
train_factor = 0.0000000000
num_workers_train = 4
num_workers_eval = 2
train_crop_size = 512
train_mask_border = 16
reduced_check_point = false
write_images_at_checkpoint = false
keep_all_scenes_in_memory = false
use_image_masks = false
write_test_images = true
texture_random_init = true
texture_color_init = false
train_use_crop = true
experiment_dir = experiments/
scene_base_dir = scenes/
scene_names = josefstaedterstrasse64 
checkpoint_directory = checkpoints/
loss_vgg = 1.0000000000
loss_l1 = 1.0000000000
loss_mse = 0.0000000000
min_zoom = 0.7500000000
max_zoom = 1.5000000000
crop_prefere_border = true
optimize_eval_camera = false
interpolate_eval_settings = false
noise_pose_r = 0.0000000000
noise_pose_t = 0.0000000000
noise_intr_k = 0.0000000000
noise_intr_d = 0.0000000000
noise_point = 0.0000000000
point_duplicate_factor = 1
lr_decay_factor = 0.7500000000
lr_decay_patience = 15
lock_camera_params_epochs = 50
lock_structure_params_epochs = 50


[RenderParams]
render_outliers = false
check_normal = true
ghost_gradients = true
drop_out_points_by_radius = true
outlier_count = 1000000
drop_out_radius_threshold = 0.6000000238
dropout = 0.2500000000
depth_accept = 0.0099999998
test_backward_mode = 0
distortion_gradient_factor = 0.0099999998
K_gradient_factor = 0.5000000000


[PipelineParams]
train = true
verbose_eval = false
log_render = true
log_texture = true
skip_neural_render_network = false
enable_environment_map = true
env_map_w = 1024
env_map_h = 512
env_map_channels = 4
num_texture_channels = 4
cat_env_to_color = false
cat_masks_to_color = false


[OptimizerParams]
texture_optimizer = adam
fix_render_network = false
fix_texture = false
fix_environment_map = false
fix_points = true
fix_poses = false
fix_intrinsics = true
fix_vignette = false
fix_response = false
fix_wb = true
fix_exposure = true
fix_motion_blur = true
fix_rolling_shutter = true
lr_render_network = 0.0004000000
lr_texture = 0.0100000000
lr_background_color = 0.0002000000
lr_environment_map = 0.0020000000
lr_points = 0.0001000000
lr_poses = 0.0050000000
lr_intrinsics = 0.0100000000
response_smoothness = 1.0000000000
lr_vignette = 0.0000050000
lr_response = 0.0010000000
lr_wb = 0.0010000000
lr_exposure = 0.0005000000
lr_motion_blur = 0.0050000000
lr_rolling_shutter = 0.0000020000


[NeuralCameraParams]
enable_vignette = true
enable_exposure = true
enable_response = true
enable_white_balance = false
enable_motion_blur = false
enable_rolling_shutter = false
response_params = 25
response_gamma = 0.4545454681
response_leak_factor = 0.0099999998


[MultiScaleUnet2dParams]
num_input_layers = 4
num_input_channels = 4
num_output_channels = 3
feature_factor = 4
num_layers = 4
add_input_to_filters = false
channels_last = false
half_float = false
upsample_mode = bilinear
norm_layer_down = id
norm_layer_up = id
last_act = 
conv_block = gated
conv_block_up = gated
pooling = average
activation = elu
```

I can only say that I tried with another scene, and I'm getting that same error.

**Comment by darglein** [comment]
Is that a HDR scene? What is the smallest and largest exposure value?

**Comment by cduguet** [comment]
I naively believe the images have exposure different enough to be HDR. 
Here's the data: 
https://drive.google.com/drive/folders/1jx_AaEDsAykUCqBAf9b7Ykuws__-U3LD?usp=sharing

**Comment by darglein** [comment]
Ok then I recomment you doing the following changes:

```
log_render = false
log_texture = false

lr_texture = 0.0800000000
lr_background_color = 0.0040000000
lr_environment_map = 0.0200000000
```

I think you based your config on `train_boat.ini` which is meant for HDR scenes.

**Comment by cduguet** [comment]
Thank you!! I'll give it a go.

**Comment by cduguet** [comment]
and yes, I based it on train_boat.

**Comment by cduguet** [comment]
It did succesfully train. I'm having problems with quality now, but I think that's for another issue. Thank you!
